packages/react-databrowser/src/components/databrowser/components/sidebar/keys-list.tsx

```diff
@@ -0,0 +1,55 @@
+import { useDatabrowserStore } from "@/store"
+
+import { cn } from "@/lib/utils"
+import { Button } from "@/components/ui/button"
+import { TypeTag } from "@/components/databrowser/components/type-tag"
+import type { RedisKey } from "@/components/databrowser/hooks"
+
+import { useKeys } from "../../hooks/use-keys"
+import { SidebarContextMenu } from "../sidebar-context-menu"
+
+export const KeysList = () => {
+  const { keys } = useKeys()
+
+  return (
+    <div className="pr-3">
+      {keys.map((data) => (
+        <KeyItem key={data[0]} data={data} />
+      ))}
+    </div>
+  )
+}
+
+const KeyItem = ({ data }: { data: RedisKey }) => {
+  const { selectedKey, setSelectedKey } = useDatabrowserStore()
+
+  const [dataKey, dataType] = data
+  const isKeySelected = selectedKey === dataKey
+
+  return (
+    <SidebarContextMenu dataKey={dataKey} key={dataKey}>
+      <Button
+        data-key={dataKey}
+        variant={isKeySelected ? "default" : "ghost"}
+        className={cn(
+          "relative flex h-10 w-full items-center justify-start gap-2 px-3 py-0 ",
+          "select-none border border-transparent text-left",
+          isKeySelected && "shadow-sm",
+          isKeySelected && dataType === "string" && "border-sky-400 !bg-sky-50 text-sky-900",
+          isKeySelected && dataType === "hash" && "border-amber-400 !bg-amber-50 text-amber-900",
+          isKeySelected && dataType === "set" && "border-indigo-400 !bg-indigo-50 text-indigo-900",
+          isKeySelected && dataType === "zset" && "border-pink-400  !bg-pink-50 text-pink-900",
+          isKeySelected && dataType === "json" && "border-purple-400 !bg-purple-50 text-purple-900",
+          isKeySelected && dataType === "list" && "border-orange-400 !bg-orange-50 text-orange-900",
+          isKeySelected && dataType === "stream" && "border-green-400 !bg-green-50 text-green-900"
```

Comment:
Instead of inlineing, we can maybe create a util function to get the necessary class